front_end/Profile/UserProfile.js

```diff
@@ -119,6 +119,14 @@ export default class UserProfile extends Component {
     }
 
     render() {
+        let curState;
+        if(this.state.isEditing){
+            curState = this.state.editState;
+        }else{
+            curState = this.state;
+        }
+        const { username, location, occupation, age } = curState;
```

Comment:
`const { whatever } = this.state.isEditing ? this.state.editState : this.state;`

The question is, why is this set up this way though? Seems like a very hacky way of structuring state